plugins/vli/fu-vli-usbhub-device.c

```diff
@@ -1006,12 +1036,25 @@ fu_vli_usbhub_device_write_firmware (FuDevice *device,
 	return FALSE;
 }
 
+static void
+fu_vli_usbhub_device_kind_changed_cb (FuVliDevice *device, GParamSpec *pspec, gpointer user_data)
+{
+	FuDeviceClass *klass_device = FU_DEVICE_GET_CLASS (device);
+	if (fu_vli_device_get_kind (device) == FU_VLI_DEVICE_KIND_VL817 &&
+	    fu_device_has_custom_flag (FU_DEVICE (device), "attach-with-gpiob"))
+		klass_device->attach = fu_vli_usbhub_device_attach_vl817_gpiob;
+}
+
```

Comment:
We think the vl817 default attach should be fu_vli_device_attach and if it has the "attach-with-gpiob" flag then evoke "fu_vli_usbhub_device_attach_vl817_gpiob". Otherwise downport hubs without share spi will not attach properly under parent; see Proj 08 Mini Dock.

Reply:
That's what I did, no? The default attach handler is set up in `_init()`.

Reply:
During testing we're running into the issue of fu_vli_usbhub_device_attach_vl817_gpiob() trying to attach with VL211 and not VL817.

Reply:
`write VL817 -> write VL103 -> vl817_gpiob` is what I'm trying to do here as a first step -- the `write VL211 -> attach VL211 -> vl817_gpiob` is still WIP.

Reply:
Makes sense.

Reply:
I've pushed a 100% untested patch to make the `write VL211 -> attach VL211 -> vl817_gpiob` sequence work.

Reply:
I've tested this, with the GPIOB toggle it seems to work on all firmware versions now.